File: sistemaFacturacion/src/main/java/prueba/sistemaFacturacion/persistence/entity/Producto.java
package prueba.sistemaFacturacion.persistence.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;

@Entity
@Table (name = "Producto")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor

public class Producto {

    @Column (name = "id_producto")
    @Id
    @GeneratedValue (strategy = GenerationType.IDENTITY)
    private Long id;


    @Column(nullable = false)
    private String nombre;


    @Column(nullable = false)
    private String descripcion;

    @Column(name = "precio_unitario", nullable = false)
    private BigDecimal precioUnitario;


    @Column(nullable = false)
    private Integer stock;


    private Boolean activo;
}

File: reactFacturacion/src/interface/producto.ts

export interface Producto {
    id: number;
    nombre: string;
    descripcion: string;
    precioUnitario: number;
    stock: number;
    activo: boolean;
}

export interface ProductoRequest {
    nombre: string;
    descripcion: string;
    precioUnitario: number;
    stock: number;
    activo: boolean;
}